src/python/WMCore/MicroService/MSPileup/MSPileupMonitoring.py

```diff
@@ -0,0 +1,110 @@
+"""
+File       : MSPileupMonitoring.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: MSPileupMonitoring provides bridge between MSPileup
+             service and CMS Monitoring infrastructure
+"""
+
+# system modules
+import time
+
+# WMCore modules
+from WMCore.MicroService.Tools.Common import getMSLogger
+
+# CMSMonitoring modules
+from CMSMonitoring.StompAMQ7 import StompAMQ7 as StompAMQ
+
+
+def flatDocuments(doc):
+    """
+    Helper function to flat out MSPileup document
+
+    :param doc: input MSPileup document
+    :return: generator of MSPileup documents flatten from original one
+    """
+    docs = flatKey(doc, 'campaigns')
+    docs = (f for d in docs for f in flatKey(d, 'currentRSEs'))
+    docs = (f for d in docs for f in flatKey(d, 'expectedRSEs'))
+    for doc in docs:
+        yield doc
+
+
+def flatKey(doc, key):
+    """
+    Helper function to flat out values of given key in a document
+
+    :param doc: input MSPileup document
+    :param key: document key to use
+    :return: generator of MSPileup documents flatten from original one and given key
+    """
+    for item in doc[key]:
+        ndoc = dict(doc)
+        # convert plural to singular key name, e.g. campaigns -> campaign
+        nkey = key[:-1]
+        ndoc[nkey] = item
+        del ndoc[key]
+        yield ndoc
+
+
+class MSPileupMonitoring():
+    """
+    MSPileupMonitoring represents MSPileup monitoring class
+    """
+
+    def __init__(self, msConfig=None):
+        """
+        Constructor for MSPileupMonitoring
+        """
+        self.userAMQ = msConfig.get('user_amq', None)
+        self.passAMQ = msConfig.get('pass_amq', None)
+        self.postToAMQ = msConfig.get('post_to_amq', False)
+        self.topicAMQ = msConfig.get('topic_amq', None)
+        self.docTypeAMQ = msConfig.get('doc_type_amg', 'cms-ms-pileup')
+        self.hostPortAMQ = msConfig.get('host_port_amq', None)
+        self.interval = msConfig.get('monitoring_interval', 600)
```

Comment:
It looks like `monitoring_interval` is actually not used. Should we remove it?

Reply:
done